packages/yoga/src/Button/native/Button.jsx

```diff
@@ -105,26 +87,58 @@ const Button = ({
   small,
   pressed,
   inverted,
+  icon: Icon,
+  theme: {
+    yoga: {
+      components: { button },
+    },
+  },
   ...rest
-}) => (
-  <ButtonContainer
-    {...rest}
-    full={full}
-    pressed={pressed}
-    disabled={disabled}
-    small={small}
-    inverted={inverted}
-  >
-    <Label
-      disabled={disabled}
+}) => {
+  let textColor = button.types.contained.font.default.color;
+
+  if (disabled) {
+    textColor = button.types.contained.font.disabled.color;
+  } else if (inverted) {
+    textColor = button.types.contained.backgroundColor.default;
+    if (pressed) {
+      textColor = button.types.contained.backgroundColor.pressed;
+    }
+  } else if (pressed) {
+    textColor = button.types.contained.font.pressed.color;
+  }
```

Comment:
It would be awesome but `font` and `backgroundColor` don't share the same state. 
If we refactor this code to a state machine, as you mentioned, we will just change this logic to another place.

This logic was inside the styled-component, in some template strings. This new approach, moving to the render, makes it much easier to see what is going on with the Button text color. [Look](https://github.com/Gympass/yoga/pull/233/files/a361a92f04b8c0f4a3321f3c66aef417ddcf35cc#diff-6a6f5acdd5d66878cb6413b9bbbe6ef72860a65b010dd4d5dc49326ee3117acbR21) what we have now and earlier and see the difference. 

I don't know if I get the point here but I'm open to discussing this implementation.

Reply:
Digging deeper into this subject and talking with you in private, it seems to be a more complex issue but a much more viable one.

Since `pressed`, `inverted`, and `disabled` are some states shared between Checkbox, Switch, and Button, to name a fews, we could export this logic and add this state as props to all components that need this. It could be a new HOC or we could take advantage of [withTouchable](https://github.com/Gympass/yoga/blob/master/packages/yoga/src/Button/native/withTouchable.jsx), for instance, or we could discuss a better approach also.

It is a great feature and should be treated as it should, in a separated PR, IMHO*

Is it good for you if we implement this as soon as we merge `alpha` into the `master` branch?




*_it could take a lot of time refactor/revisiting the components_